src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/TransportChangePolicyAction.kt

```diff
@@ -0,0 +1,297 @@
+import org.elasticsearch.ElasticsearchStatusException
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest
+import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse
+import org.elasticsearch.action.bulk.BulkRequest
+import org.elasticsearch.action.bulk.BulkResponse
+import org.elasticsearch.action.get.GetRequest
+import org.elasticsearch.action.get.GetResponse
+import org.elasticsearch.action.get.MultiGetRequest
+import org.elasticsearch.action.get.MultiGetResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.action.support.IndicesOptions
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+
+private val log = LogManager.getLogger(TransportChangePolicyAction::class.java)
+
+class TransportChangePolicyAction @Inject constructor(
+    val client: NodeClient,
+    transportService: TransportService,
+    actionFilters: ActionFilters,
+    val clusterService: ClusterService,
+    val xContentRegistry: NamedXContentRegistry
+) : HandledTransportAction<ChangePolicyRequest, ISMStatusResponse>(
+    ChangePolicyAction.NAME, transportService, actionFilters, ::ChangePolicyRequest
+) {
+    override fun doExecute(task: Task, request: ChangePolicyRequest, listener: ActionListener<ISMStatusResponse>) {
+        ChangePolicyHandler(client, listener, request).start()
+    }
+
+    inner class ChangePolicyHandler(
+        private val client: NodeClient,
+        private val actionListener: ActionListener<ISMStatusResponse>,
+        private val request: ChangePolicyRequest
+    ) {
+        private val failedIndices = mutableListOf<FailedIndex>()
+        private val managedIndexUuids = mutableListOf<Pair<String, String>>()
+        private val indexUuidToCurrentState = mutableMapOf<String, String>()
+        lateinit var policy: Policy
+        lateinit var response: GetResponse
+        private var updated: Int = 0
+
+        fun start() {
+            val getRequest = GetRequest(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX, request.changePolicy.policyID)
+
+            client.get(getRequest, object : ActionListener<GetResponse> {
```

Comment:
Did this not work? `client.get(getRequest, ActionListener.wrap(::onGetPolicyResponse, ::onFailure))`